src/main/java/build/buildfarm/instance/shard/RedisShardSubscription.java

```diff
@@ -87,14 +87,17 @@ private void mainLoop() throws IOException {
     }
   }
 
-  public void stop() {
+  public void stop() throws JedisException {
     if (stopped.compareAndSet(false, true)) {
       try {
         subscriber.unsubscribe();
       } catch (JedisException e) {
-        // subscriber validates with private member to determine connection status
-        // detect this condition and ignore it
-        if (!e.getMessage().endsWith(" is not connected to a Connection.")) {
+        // If stop() is called before a connection is established, log and throw the exception
+        if (e.getMessage().endsWith(" is not connected to a Connection.")) {
```

Comment:
You're swallowing any JedisException other than the "is not connected" exception, which deviates from the previous behavior.

Reply:
I'll move the `throw` outside of the `if` block so all are thrown but the "no connection" ones get the log explaining.